x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/type/KeywordEsField.java

```diff
@@ -27,12 +29,11 @@ public KeywordEsField(String name, Map<String, EsField> properties, boolean hasD
     
     public KeywordEsField(String name, Map<String, EsField> properties, boolean hasDocValues, int precision,
             boolean normalized, boolean isAlias) {
-        super(name, DataType.KEYWORD, properties, hasDocValues, isAlias);
+        super(name, KEYWORD, properties, hasDocValues, isAlias);
         this.precision = precision;
         this.normalized = normalized;
     }
 
-    @Override
     public int getPrecision() {
```

Comment:
What does the `precision` mean in the context of a generic QL keyword field?

Reply:
Being a string, it represents the number of characters - namely `ignore_above`.